test/all-templates.test.js

```diff
@@ -0,0 +1,42 @@
+'use strict';
+
+// This file contains Jest tests that run against every function template in
+// the repo.
+
+const path = require('path');
+const fs = require('fs');
+
+const projectRoot = path.resolve(__dirname, '..');
+// Assemble a list of template directories here, since templates.json
+// may not have all of them:
+const templates = fs
+      .readdirSync(projectRoot, { withFileTypes: true })
+      .filter(
+          (file) =>
+          file.isDirectory() &&
+              !file.name.startsWith(".") &&
+              !file.name.startsWith("_") &&
+              !["node_modules", "test", "coverage", "docs"].includes(file.name)
+      )
+      .map((dir) => dir.name)
+      .filter(
+        (template) => {
+          try {
+            const packageJson = path.join(projectRoot, template, 'package.json');
```

Comment:
what impact does the presence of the `package.json` have on curating a list of templates? How does this impact a check on the .env file?

Reply:
I use `package.json` as something that should always exist in a function template subdirectory, since I believe all of the current mechanisms for function deployment expect one. I use it to exclude directories that are definitely not deployable as function templates, since switching branches with `git checkout` can leave partial function template directories behind (with `.gitignore`ed files inside the directories) if you're moving from a branch where that template exists to one where it doesn't; without the `package.json` filter, switching branches can therefore cause `npm test` to fail in a way that isn't directly related to the commit at `HEAD`, which felt surprising to me.